apps/api/src/tasks/bank-import/import-transactions.task.ts

```diff
@@ -309,8 +333,8 @@ export class IrisTasks {
         senderIban: trx.debtorAccount?.iban?.trim(),
         recipientIban: trx.creditorAccount?.iban?.trim(),
         type: trx.creditDebitIndicator === 'CREDIT' ? 'credit' : 'debit',
-        amount: toMoney(trx.transactionAmount?.amount),
-        currency: trx.transactionAmount?.currency,
+        amount: toMoney(transactionAmount.amount),
+        currency: transactionAmount.currency,
         description: trx.remittanceInformationUnstructured?.trim(),
         // Not saved in the DB, it's added only for convinience and efficiency
         matchedRef: matchedRef ? matchedRef[0] : null,
```

Comment:
@igoychev  If we don't parse the transaction amount (i.e. it's a valid number > 0) I've left it to not change the transaction amount and currency at all. I thought this was a good way to protect us from handling cases that we're not sure how to handle. My concern is that it might be a valid transaction, but the amount is not in the id where expect it.

Let me know what you think?

Reply:
> If we don't parse the transaction amount (i.e. it's a valid number > 0) I've left it to not change the transaction amount and currency at all. I thought this was a good way to protect us from handling cases that we're not sure how to handle.

Yes, this is ok, just for this case, we also need to set the matchedRef to null so that the record is considered as unrecognized

Reply:
While writing a test I hit a corner case here. If it's EUR + not recognized, as per a previous comment we always set "BGN" as Currency and the amount would be the only one we have - the EUR amount, and it will be confusing.
So I'd suggest that we leave the 'BGN' currency override only when we can parse it and leave the original currency + mark as unrecognized in the other case?

Reply:
yes, that's the idea - leave everything as it came

Reply:
Ok it's updated now.